website/docs/r/nas_datastore.html.markdown

```diff
@@ -56,33 +56,33 @@ resource "vsphere_nas_datastore" "datastore" {
 
 The following arguments are supported:
 
-* `name` - (String, required, forces new resource) The name of the datastore.
-* `host_system_ids` - (List of strings, required) The managed object
-  IDs of the hosts to mount the datastore on.
-* `folder` - (String, optional) The relative path to a folder to put this
-  datastore in. This is a path relative to the datacenter you are deploying the
-  datastore to. Example: for the `dc1` datacenter, and a provided `folder` of
-  `foo/bar`, Terraform will place a datastore named `terraform-test` in a
-  datastore folder located at `/dc1/datastore/foo/bar`, with the final
-  inventory path being `/dc1/datastore/foo/bar/terraform-test`.
-* `type` - (String, optional, forces new resource) The type of NAS volume. Can
-  be one of `NFS` (to denote v3) or `NFS41` (to denote NFS v4.1). Default:
-  `NFS`.
-* `remote_hosts` - (List of strings, required, forces new resource) The
-  hostnames or IP addresses of the remote server or servers. Only one element
-  should be present for NFS v3 but multiple can be present for NFS v4.1.
-* `remote_path` - (String, required, forces new resource) The remote path of
-  the mount point.
-* `access_mode` - (String, optional, forces new resource) Access mode for the
-  mount point. Can be one of `readOnly` or `readWrite`. Note that `readWrite`
-  does not necessarily mean that the datastore will be read-write depending on
-  the permissions of the actual share. Default: `readWrite`.
-* `security_type` - (String, optional, forces new resource) The security type
-  to use when using NFS v4.1. Can be one of `AUTH_SYS`, `SEC_KRB5`, or
-  `SEC_KRB5I`.
-* `tags` - (List of strings, optional) The IDs of any tags to attach to this
-  resource. See [here][docs-applying-tags] for a reference on how to apply
-  tags.
+* `name` - (Required) The name of the datastore. Forces a new resource if
+  changed.
+* `host_system_ids` - (Required) The managed object IDs of the hosts to mount
```

Comment:
I think this is possibly a domain knowledge issue. Since most work in vSphere is done in the GUI with human names, IDs are rarely exposed. In the API though, every managed object is has an ID, which depending on the literature you happen to be reading is either MoRef, or MOID, but not necessarily ID itself. We shorten here to be terse of course, but I think in long form it's okay just having "managed object ID" to be specific on what kind of ID we are expecting here.

What might be a better idea is explaining how the vSphere provider now largely deals in MOIDs and UUIDs. This has been an intentional move away from names (moving these to data sources instead) to allow for better search flexibility and to minimize spurious diffs in downstream resources (and any necessary normalization logic) due to possible different combinations of offering a specific object's inventory path.

Will add that in the index.